app/src/main/java/com/android/sample/ui/authentication/SignIn.kt

```diff
@@ -102,52 +100,70 @@ fun SignInScreen(navigationActions: NavigationActions, viewModel: SignInViewMode
-                          else -> {
-                            passwordErrorState.value =
-                                null // Clear external error if password is valid
-                            viewModel.signInWithEmailAndPassword(
-                                emailState.value,
-                                passwordState.value,
-                                onProfileExists,
-                                onProfileMissing,
-                                onSignInFailure)
-                          }
-                        }
-                        Log.d("SignInScreen", "Sign in with email/password")
-                      })
-                },
-                modifier =
-                    Modifier.fillMaxWidth(WIDTH_FRACTION)
-                        .height(BUTTON_HEIGHT.dp)
-                        .testTag("SignInButton")) {
-                  Text("Sign in with Email", fontSize = SUBTITLE_FONTSIZE.sp)
+            Card(
+                modifier = Modifier.fillMaxWidth(0.7f).testTag("SignInCard"),
+                colors =
+                    CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.background),
+                elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
+                shape = RoundedCornerShape(12.dp)) {
+                  Button(
+                      onClick = {
+                        performOfflineAwareAction(
+                            context = context,
+                            networkManager = networkManager,
+                            onPerform = {
+                              when {
+                                !isValidEmail(emailState.value) -> {
+                                  emailErrorState.value =
+                                      "Please enter a valid address: example@mail.xx"
+                                }
+                                passwordState.value.isEmpty() -> {
+                                  passwordErrorState.value =
+                                      "Password cannot be empty" // Set external error
+                                }
+                                else -> {
+                                  passwordErrorState.value =
+                                      null // Clear external error if password is valid
+                                  viewModel.signInWithEmailAndPassword(
+                                      emailState.value,
+                                      passwordState.value,
+                                      onProfileExists,
+                                      onProfileMissing,
+                                      onSignInFailure)
+                                }
+                              }
+                              Log.d("SignInScreen", "Sign in with email/password")
+                            })
+                      },
+                      modifier = Modifier.fillMaxWidth().height(60.dp).testTag("SignInButton"),
```

Comment:
Here you have some magic values, please correct this thanks

Reply:
Done